### PR TITLE
Added 2 blog tutorials under AWS + OpenShift blog too

### PR DESCRIPTION
**Pull request by ajeetraina**

### CODE REVIEW

**Comment by ajeetraina** [comment]
@simonprickett Can you help me with the review?

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+You’ll build an application that will allow users to attend and take part in digital auctions. The application will allow users to create an account, put in bids, and even set up their own auction. Below we’ll uncover the required components, their functionality, and how to deploy them within this architecture.
+
+Ready to get started? Ok, let’s dive straight in.
+
+### 2. What will you need?
+
+- [NodeJS](https://developer.redis.com/develop/node): used as an open-source, cross-platform, backend JavaScript runtime environment that executes Javascript code outside a web browser.
+- [Amazon Cognito](https://aws.amazon.com/cognito/): used to securely manage and synchronize app data for users on mobile.
+- [Redis Enterprise Cloud](https://developer.redis.com/create/rediscloud): used as a real-time database, cache, and message broker.
+- [RedisJSON](https://developer.redis.com/howtos/redisjson/getting-started): used to store, update and fetch JSON values from Redis keys.
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx]: "used to store, update and fetch JSON values from Redis ~keys~."

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+- [Socket.IO](https://socket.io/): used as a library that provides real-time, bi-directional, and event-based communication between the browser and the server.
+- [AWS Lambda](https://aws.amazon.com/lambda/): used a serverless compute service that runs your code in response events and manages the underlying compute service automatically for you.
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx]: "used **as** a"

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+npm install is a npm cli-command that does the predefined thing i.e install dependencies specified inside package.json
+
+```
+npm install
+```
+
+### Step 3. Building the application
+
+```
+npm build
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx]: npm run build

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+
+### Step 10. Accessing the bidding page
+
+![accessing](images/image_7.png)
+
+
+### 6. How data is stored
+
+The Redis Enterprise  Cloud Database with  RedisJSON module is what you’ll use to install the data.
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx]: Remove the extra spaces here

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+### Auctions
+
+- Type - Redis Hash.
+- Used for storing auctions data.
+- UUID generated from the backend (NodeJS) serves as the key
+- JSON data which includes keys - auctionId, auctionItemName, description,lotNo, quantity, buyersPremium, itemUnit, minBidAmount, bidIncrement, startDateTime, endDateTime, images, currentBid - servers as the value for Auctions hash.
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx]: - JSON data which represents the Auction object and includes the following keys
  - auctionId
  - auctionItemName
  - description
  - lotNo
  - quantity
  - buyersPremium
  - itemUnit
  - minBidAmount
  - bidIncrement
  - startDateTime
  - endDateTime
  - images
  - currentBid

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+- NodeJS connects to the Redis Cloud database. Meanwhile, the frontend communicates with the NodeJS backend through API calls.
+- POST : /api/auctions.
+- The request body has JSON data to be inserted into the database.
+- NodeJS uses the Redis module to work with Redis Cloud. The Redis client is created. using the Redis credentials and hmset(). This is the equivalent of the HMSET command that’s used to push data to the Redis database.
+
+### Biddings
+
+- Type - Redis Hash
+- Used for storing the bids placed on each auction item
+- NodeJS connects to the Redis Cloud database. The Frontend communicates with the NodeJS backend through API calls.
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx:208]: ~F~frontend

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+- POST : ```/api/bidding```
+- The request body has JSON data to be inserted into the database.
+- AuctionId from request body servers as the key
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx]: serve~r~s

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+- JSON data which includes keys - currentBid, currentBidTime, currentBidEndTime, and biddings array (id, auctionId, userId, username, bidAmount, bidTime) - servers as value
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx]: serve~r~s

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+- NodeJS connects to the Redis Cloud database. The frontend communicates with the NodeJS backend through API calls.
+- POST : ```/api/settings```
+- The request body has JSON data to be inserted into the database.
+- NodeJS uses the Redis module to work with Redis Cloud. The Redis client is created using the Redis credentials and hmset(). This is the equivalent of the HMSET command that’s used to push data to the Redis database.
+
+### Users
+
+- Type - Redis Hash
+- Used for storing user details
+- NodeJS connects to the Redis Cloud database. The Frontend communicates with the NodeJS backend through API calls.
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx:238]: ~F~frontend

File: docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+
+But thanks to Redis, the components that made up the architecture system became vastly more interconnected so data was able to be sent, processed, and received in real-time. Achieving this paves the way for a smooth bidding process where users can interact with events in real-time without interruptions, ultimately enhancing the functionality of the app.
+
+[NR-Digital-Auction](https://launchpad.redis.com/?id=project%3ANR-digital-auction-frontend) is a fantastic example of how innovations can be brought to life by using Redis. Everyday programmers are experimenting with Redis to build applications that are impacting everyday life from around the world and you can too!
+
+So what can you build with Redis? For more inspiration, you can head over to the Redis Launchpad to access an exciting range of applications.
+
+Check it out. Be inspired. And have fun with Redis.
+
```
Comment (wjohnsto) [inline docs/create/aws/bidding-on-aws/index-bidding-on-aws.mdx:297]: Add the MATRIX200 promo here

File: docs/create/aws/chatapp/index-chatapp.mdx [hunk 11]
```diff
@@ -0,0 +1,262 @@
...
+sidebar_label: Building a Real-Time Chat application on AWS using Python and Redis
+slug: /create/aws/chatapp
+authors: [ajeet]
+---
+
+Chat messaging apps are surging in popularity exponentially. Mobile apps like WhatsApp, Facebook, Telegram, Slack, Discord have become “a part and parcel” of our life. Users are addicted to these apps as they bring  a personal touch and offer a real-time interaction.
+
+![chatapp](image_chatapp1.png)
+
+There’s been a rise in the number of social media apps that bring social elements to enable activities like collaboration, messaging, social interaction and commenting. Such activities require a real-time capability to automatically present updated information to users. More and more developers are tapping into the power of Redis as it is extremely fast & due to its support for a variety of rich data structures such as Lists, Sets, Sorted Sets, Hashes etc. Redis comes along with a Pub/Sub messaging functionality that allows developers to scale the backend by spawning multiple server instances.
```
Comment (wjohnsto) [inline docs/create/aws/chatapp/index-chatapp.mdx]: ~& due to its support for~ and it supports

File: docs/create/aws/chatapp/index-chatapp.mdx [hunk 12]
```diff
@@ -0,0 +1,262 @@
...
+ ```
+
+Additionally, each user is added to the default "General" room. 
+For handling rooms for each user, we have a set that holds the room ids. Here's an example command of how to add the room: 
+
+ ```bash
+  SADD user:1:rooms "0"
+ ```
+
+Populate private messages between users. At first, private rooms are created: if a private room needs to be established, for each user a room id: room:1:2 is generated, where numbers correspond to the user ids in ascending order.
```
Comment (wjohnsto) [inline docs/create/aws/chatapp/index-chatapp.mdx]: Is there supposed to be a heading here? The first sentence doesn't make sense.

File: docs/create/aws/chatapp/index-chatapp.mdx [hunk 13]
```diff
@@ -0,0 +1,262 @@
...
+ ```bash
+  SADD user:1:rooms 1:2 and SADD user:2:rooms 1:2
+ ```
+
+Then we add messages to this room by writing to a sorted set: 
+
+ ```bash
+  ZADD room:1:2 1615480369 "{'from': 1, 'date': 1615480369, 'message': 'Hello', 'roomId': '1:2'}"
+ ```
+We use a stringified JSON for keeping the message structure and simplify the implementation details for this demo-app.
```
Comment (wjohnsto) [inline docs/create/aws/chatapp/index-chatapp.mdx]: We are using a stringified JSON to keep the message structure and simplify the implementation details for this demo-app. You may choose to use a Hash or RedisJSON.

File: docs/create/aws/chatapp/index-chatapp.mdx [hunk 14]
```diff
@@ -0,0 +1,262 @@
...
+
+After initialization, a pub/sub subscription is created: SUBSCRIBE MESSAGES. At the same time, each server instance will run a listener on a message on this channel to receive real-time updates.
+
+Again, for simplicity, each message is serialized to JSON, which we parse and then handle in the same manner, as WebSocket messages.
+
+Pub/sub allows connecting multiple servers written in different platforms without taking into consideration the implementation detail of each server.
+
+#### Real-time chat and session handling
+
+When a WebSocket/real-time server is instantiated, which listens for the next events:
```
Comment (wjohnsto) [inline docs/create/aws/chatapp/index-chatapp.mdx]: When a WebSocket connection is established, we can start to listen for events:

**Review by wjohnsto — APPROVED** [review]
Looks good once you implemented the suggested changes!

File: docs/create/aws/chatapp/index-chatapp.mdx (hunk 12 above)
Comment (ajeetraina) [reply docs/create/aws/chatapp/index-chatapp.mdx]: Yes, let me fix this.

**Pull request merged**